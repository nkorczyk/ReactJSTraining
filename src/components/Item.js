import React from 'react';

const Item = ({ movie }) => {
  const genres = movie.genres.map((genre, index, genres) => (
    <span key={index}>{genre} & </span>
  ));

  return (
    <div>
      <img src={movie.poster_path} className="thumbnails" alt="" />
      <div className="details">
        <span className="movie-title">{movie.title.toUpperCase()}</span>
        <span className="release-date">{movie.release_date}</span>
      </div>
      <span className="genre">{genres}</span>
    </div>
  )
}

export default Item;
