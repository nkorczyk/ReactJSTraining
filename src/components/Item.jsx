// @flow
import React from 'react';
import { Link } from "react-router-dom";
import styled from 'styled-components';

type Props = {
    movie: {
      id: string,
      poster_path: string,
      release_date: string,
      overview: string,
      title: string,
      genres: Array<string>,
    }
}

const Thumbnails = styled.img`
  display: block;
  width: 256px;
  cursor: pointer;
`;

const MovieTitle = styled.span`
  float: left;
  font-size: 10px;
  font-weight: bold;
  padding: 2px 2px;
`;

const ReleaseDate = styled.span`
  float: right;
  font-size: 10px;
  padding: 0px 1px;
  margin-top: 1px;
  border: 1px solid #000;
`;

const Genre = styled.span`
  font-size: 10px;
`;

const Item = (props: Props) => {

  const { id, poster_path, overview, title, release_date } = props.movie;
  const genres = props.movie.genres.join(" & ");

  return (
    <div>
      <Link to={`/film/${id}`}>
        <Thumbnails src={poster_path} alt={overview} />
      </Link>
      <div className="details">
        <MovieTitle>{title.toUpperCase()}</MovieTitle>
        <ReleaseDate>{release_date}</ReleaseDate>
      </div>
      <Genre>{genres}</Genre>
    </div>
  )
}

export default Item;
