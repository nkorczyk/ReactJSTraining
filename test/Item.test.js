import React from 'react';
import { shallow } from 'enzyme';
import Item from '../src/components/Item';

describe('should render Item component', () => {
  let element;
  beforeAll(() => {
    const movies_data = {
      id: 321612,
      title: 'Beauty and the Beast',
      tagline: 'Be our guest.',
      vote_average: 6.8,
      vote_count: 7861,
      release_date: '2017-03-16',
      poster_path: 'https://image.tmdb.org/t/p/w500/tWqifoYuwLETmmasnGHO7xBjEtt.jpg',
      overview: "A live-action adaptation of Disney's version",
      genres: [
        'Family',
        'Fantasy',
        'Romance',
      ],
    };
    element = shallow(<Item movie={movies_data} />);
  });
  it('Snapshot test with default props', () => {
    expect(element).toMatchSnapshot();
  });
});
