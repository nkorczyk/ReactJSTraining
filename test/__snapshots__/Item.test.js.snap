// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Item component Snapshot test with default props 1`] = `
<div>
  <Link
    replace={false}
    to="/film/321612"
  >
    <styled.img
      alt="A live-action adaptation of Disney's version"
      src="https://image.tmdb.org/t/p/w500/tWqifoYuwLETmmasnGHO7xBjEtt.jpg"
    />
  </Link>
  <div
    className="details"
  >
    <styled.span>
      BEAUTY AND THE BEAST
    </styled.span>
    <styled.span>
      2017-03-16
    </styled.span>
  </div>
  <styled.span>
    Family & Fantasy & Romance
  </styled.span>
</div>
`;
